moped-editor/src/views/projects/projectView/ProjectTeam/ProjectTeamTable.js

```diff
@@ -0,0 +1,506 @@
+        name: null,
+        moped_workgroup: null,
+        moped_proj_personnel_roles: [],
+        notes: null,
+        isNew: true,
+        roleIds: [],
+        project_personnel_id: id,
+      },  
+      ...oldRows,
+    ]);
+    setRowModesModel((oldModel) => ({
+      ...oldModel,
+      [id]: { mode: GridRowModes.Edit, fieldToFocus: "project_personnel_id" },
+    }));
+  }
+
+  const handleEditClick = useCallback((id) => () => {
+    setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.Edit } });
+  }, [rowModesModel]);
+
+  const handleSaveClick = useCallback((id) => () => {
+    setRowModesModel({ ...rowModesModel, [id]: { mode: GridRowModes.View } });
+  }, [rowModesModel]);
+
+  const handleCancelClick = (id, tableId) => () => {
+    setRowModesModel({
+      ...rowModesModel,
+      [id]: { mode: GridRowModes.View, ignoreModifications: true },
+    });
+    const editedRow = rows.find((row) => row[tableId] === id);
+    if (editedRow.isNew) {
+      setRows(rows.filter((row) => row.id !== id));
+    }
+  };
+
+  const handleDeleteOpen = useCallback((id) => {
+    setDeleteTeamMemberId(id);
+    return deleteProjectPersonnel({
+      variables: { id },
+    }).then(() => refetch()); 
+  }, [deleteTeamMemberId]);
+
+  const processRowUpdate = useCallback((updatedRow, originalRow, params, data) => {
+    let userId;
+    const userObject = data.moped_users.find(user => {
+      if (typeof updatedRow.moped_user === 'string') {
+        return `${user.first_name} ${user.last_name}` === updatedRow.moped_user;
+      } else {
+        return user.user_id === updatedRow.moped_user.user_id;
+      }
+    });
+
+    if (userObject) {
+      userId = userObject.user_id;
+      updatedRow.moped_user = userObject; // Update with full user object
+    } else {
+      console.error('Invalid user data:', updatedRow.moped_user);
+      throw new Error('Invalid user data');
+    }
+
+    if (updatedRow.isNew) {
+      const payload = getNewPersonnelPayload({ newData: updatedRow, projectId });
+
+      return insertProjectPersonnel({
+        variables: {
+          object: payload,
+        },
+      })
+      .then(() => refetch())
+      .then(() => updatedRow)
+      .catch((error) => {
+        console.error('Mutation error:', error);
+        throw error;
+      });
+    } else {
+          // Ensure project_personnel_id is an integer
+    const personnelId = parseInt(updatedRow.project_personnel_id);
+
+    if (!personnelId) {
+      console.error('Invalid project_personnel_id:', updatedRow.project_personnel_id);
+      throw new Error('Invalid project_personnel_id');
+    }
+
+
+
+
+    // Update roleIds to be in sync with moped_proj_personnel_roles
+    updatedRow.roleIds = updatedRow.moped_proj_personnel_roles.map(
+      (role) => role.project_role_id
+    );
+
+    // // Check if the roles have changed
+    // const haveRolesChanged = !isEqual(updatedRow.moped_proj_personnel_roles, originalRow.moped_proj_personnel_roles);
```

Comment:
~I tested changing the roles, and I see that it is working so I think this code can go. Can you remove?~

I tested creating a new team member, and I can choose roles, but I am not seeing the roles that are selected being sent in the insert payload. Could you check on this part and see if you can replicate?

Reply:
I found that the roles show up when editing a row, but not when adding a new team member.